src/pages/ProductList/Detail/ProductCartSide.js

```diff
@@ -0,0 +1,174 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+
+const ProductCartSide = props => {
+  const { name, price } = props;
+  const [count, setCount] = useState(1);
+  const navigate = useNavigate();
+
+  const onCart = () => {
+    //TODO:장바구니 담는 로직(fetch) api 입력
+    fetch('http://10.58.52.128:3000/carts', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        name: name,
+        quantity: count,
+        price: price,
+      }),
+    })
+      .then(response => response.json())
+      .then(data => {
+        if (data.message === 'SUCCESS') {
+          if (
+            window.confirm(
+              '장바구니에 담았습니다. 장바구니로 이동하시겠습니까?'
+            )
+          ) {
+            navigate('/cart');
+          }
+        } else {
+          alert('오류가 발생했습니다!');
+        }
+      });
```

Comment:
```suggestion javascript
.then(() => {
  if (response.ok === true) {
    if (window.confirm("장바구니에 담았습니다. 장바구니로 이동하시겠습니까?")) {
      navigate("/cart");
    } else return;
  }
})
```

* .then을 무조건 2개 써야 하는건 아닙니다.
* 이 코드는 장바구니로 보냈는지, 안보냈는지만 판단하면 되는 구조네요.
* 첫번째 then에서 응답받은 결과를 가지고 뭔가 액션을 취하지 않아도됩니다.
* 그래서 두번째 then은 없어도 되고,
* 분기처리는 첫번째 then에서 응답받은 `response.ok가 true`인지, 혹은 `response.status가 200번대` 라는 조건을 가지고 분기처리 하면되겠습니다.
* 만약 response.ok가 false, 혹은 response.status가 400번대 이거나, 500번대라면 응답에 맞는 처리를 해주면 되겠습니다.
* 그리고, confirm에서 반환된 값이 false일 때 오류가 발생했다고 한건 어???!!! 하고 당황할것같아요 ㅋㅋ
* 장바구니로 이동하지않았을 뿐인데 오류...?!
* 그래서 return으로 아무런 동작도 하지않고 끝내버리는 방법이 있습니다...